### PR TITLE
[Reporting] Updating reporting tests to be sca compliant

### PR DESCRIPTION
**Pull request by ColeHiggins2**
SCA is now the default in 6.16. This PR is for updating all tests under the Reporting Component
SAT-24355

### CODE REVIEW

File: tests/foreman/api/test_reporttemplates.py [hunk 1]
```diff
@@ -811,8 +715,11 @@ def test_positive_generate_job_report(setup_content, module_target_sat, content_
             'input_values': {"job_id": job["id"]},
         }
     )
-    assert res[0]['Host'] == content_hosts[0].hostname
-    assert res[1]['Host'] == content_hosts[1].hostname
+    assert (
+        res[0]['Host'] == content_hosts[0].hostname and res[1]['Host'] == content_hosts[1].hostname
+    ) or (
+        res[0]['Host'] == content_hosts[1].hostname and res[1]['Host'] == content_hosts[0].hostname
+    )
```
Comment (vsedmik) [inline tests/foreman/api/test_reporttemplates.py:722]: Let's use sets to check if all hosts were reported. 
```suggestion
    assert {i.Host for i in res} == {i.hostname for i in content_hosts}
```

**Review by vsedmik — CHANGES_REQUESTED** [review]
Looks good, just one suggestion for improvement.
Could you run PRT for some test using the `setup_content` fixture? The `test_positive_generate_job_report` could be a good candidate.

**Comment by ColeHiggins2** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_reporttemplates.py::test_positive_generate_job_report

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7278
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/api/test_reporttemplates.py::test_positive_generate_job_report --external-logging
Test Result : ================ 6 passed, 1060 warnings in 3358.02s (0:55:58) =================
```

**Pull request merged**